### ISSUE TITLE
Should `any_is_na_linter()` be extended to cover `sum(is.na(x)) != 0L`?

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1575]
It doesn't seem that [uncommon](https://github.com/search?q=%22sum%28is.na%22&type=code) of a pattern to find `NA`s.

``` r
library(lintr)

x <- rep(c(TRUE, NA, FALSE, FALSE), 1000)

identical(
  sum(is.na(x)) != 0L, 
  anyNA(x)
)
#> [1] TRUE

# currently doesn't lint
lint(
  text = "sum(is.na(x)) != 0L",
  linters = any_is_na_linter()
)
```

<sup>Created on 2022-09-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
I would put this is under our BooleanArithmeticLinter, which covers other
logical aggregations passed to sun() instead of any/all

**Comment by IndrajeetPatil** [comment]
I don't see `BooleanArithmeticLinter ` in https://github.com/r-lib/lintr/issues/884, but maybe it's not included there. 

Sounds good to me.

**Comment by MichaelChirico** [comment]
I haven't been updating the list in #884 but we have developed a few more
in the meantime 😅

**Comment by IndrajeetPatil** [comment]
So exciting!

![happy-smile](https://user-images.githubusercontent.com/11330453/193330311-b087c0c7-6146-49fe-9f9e-7f295f83911c.gif)

**Comment by MichaelChirico** [comment]
Subsuming this under #1580
